src/models/modelUtils.js

```diff
@@ -24,3 +24,10 @@ export const validImageURL = (val) => {
 
     return regex.test(val);
 };
+
+export const validApplyURL = (val) => {
+    const httpRegex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)?/;
+    const emailRegex = /mailto:(\S+@\S+)/;
+
+    return httpRegex.test(val) || emailRegex.test(val);
```

Comment:
This is not enough as it seems.

![image](https://user-images.githubusercontent.com/23351316/168555640-6af3bf9d-0b92-4aa2-8fbb-f8c16d9b5683.png)

You can use something like this and bypass this security check:

![image](https://user-images.githubusercontent.com/23351316/168555764-64dbb713-d9a7-4381-88e0-4868e4d2f188.png)

Add another additional test that checks this behaviour

Reply:
I think it's missing the start of line basically (`^https`). All it should check is to start with "http", "https" or "mailto", and I believe that the browser/React should be able to handle the rest (for example `"` escaping should not be possible even if the regex does not filter this out, I think).

Reply:
For example updating the regex to `httpRegex = /^https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)?$/;` already adresses this issue.

Reply:
For example, I was doing some testing and I think that something as simple as `httpRegex = /^https?:\/\/[a-zA-Z0-9#&=/.]{1,343}$/;` will probably work for our use-case. Given that the aim is simply to filter out potential XSS attempts, just focus on that specifically, the rest is not relevant in this case.

Reply:
I agree, that makes sense. We'll probably also have a confirmation in the FE letting the user know where he's getting redirected to

Reply:
Yes, the frontend will have a prompt informing the user that he will be redirected

Reply:
@miguelpduarte Now that I'm testing that regex, shouldn't we be only more inclusive? That one excludes query params like `?a=1&b=2` and other valid characters that were present in the first expression. I'll try to simplify the regex without making more strict

Reply:
I've got one more question. Should we allow the companies to submit URLs without the http/https? e.g. 'www.google.com`